components/dashboard/src/components/RepositoryFinder.tsx

```diff
@@ -4,137 +4,95 @@
+                setSuggestedContextURLs(urls);
+                saveSearchData(urls);
+            });
+    }, [suggestedContextURLs]);
+
+    const getElements = useMemo(
+        () => (searchString: string) => {
+            const result = [...suggestedContextURLs];
+            try {
+                // If the searchString is a URL, and it's not present in the proposed results, "artificially" add it here.
+                new URL(searchString);
+                if (!result.includes(searchString)) {
+                    result.push(searchString);
+                }
+            } catch {}
+            return result
+                .filter((e) => e.toLowerCase().indexOf(searchString.toLowerCase()) !== -1)
+                .map(
+                    (e) =>
+                        ({
+                            id: e,
+                            element: (
+                                <div className="flex-col ml-1 mt-1 flex-grow">
+                                    <div className="flex">
+                                        <div className="text-gray-700 font-semibold">
+                                            {e.substring(e.indexOf("//") + 2)}
+                                        </div>
+                                        <div className="ml-1 text-gray-400">{}</div>
+                                    </div>
+                                    <div className="flex text-xs text-gray-400">{}</div>
+                                </div>
+                            ),
+                            isSelectable: true,
+                        } as DropDown2Element),
+                );
+        },
+        [suggestedContextURLs],
+    );
 
     return (
-        <form onSubmit={onSubmit}>
-            <div className="flex px-4 rounded-xl border border-gray-300 dark:border-gray-500">
-                <div className="py-4">
-                    <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 16 16" width="16" height="16">
-                        <path
-                            fill="#A8A29E"
-                            d="M6 2a4 4 0 100 8 4 4 0 000-8zM0 6a6 6 0 1110.89 3.477l4.817 4.816a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 010 6z"
-                        />
-                    </svg>
+        <DropDown2
+            getElements={getElements}
+            onSelectionChange={props.setSelection}
+            searchPlaceholder="Paste repository URL or type to find suggestions"
+        >
+            <div className="flex h-12" title="Repository">
+                <div className="mx-2 my-2">
+                    <img className="w-8 filter-grayscale self-center" src={Repository} alt="logo" />
+                </div>
+                <div className="flex-col ml-1 mt-1 flex-grow">
+                    <div className="flex font-semibold text-gray-700">
+                        <div>Repository</div>
+                    </div>
+                    <div className="flex text-xs text-gray-500 font-semibold">
```

Comment:
**question**: Could we use a lighter black for this when there's no repository selected to indicate this is not set? Using `bg-gray-400` and `dark:bg-gray-500` could suffice.

| BEFORE | AFTER |
|-|-|
| <img width="341" alt="set-before" src="https://user-images.githubusercontent.com/120486/208642243-21fdf463-1087-4557-9b55-a45339383d67.png"> | <img width="341" alt="set-after" src="https://user-images.githubusercontent.com/120486/208642247-09821435-86c5-492d-88c7-a130dc01fef3.png"> |
| <img width="341" alt="set-dark-before" src="https://user-images.githubusercontent.com/120486/208642990-09aba311-f70b-4bc5-a14b-aca4ca51b77e.png"> | <img width="341" alt="set-dark-after" src="https://user-images.githubusercontent.com/120486/208642995-94024d53-d7f7-4db0-ba63-3a0d506c17e2.png"> |